Makefile

```diff
@@ -16,3 +16,9 @@ run:
 test: ruff pylint
 	export PYTHONPATH=src/main/py; \
 	python3 -m unittest discover --pattern "*_test.py" --start-directory src/test/py --top-level-directory .
+
+clean:
+	@echo "I'm cleaning OwO"
+	rm -rf venv
+	find . -name "*.pyc" -delete
+	find . -type d -name "__pycache__" -delete
```

Comment:
I would limit this to `src/` and condense it to:
`find src/ -print0 -type d -name "__pycache__" | xargs -0 rm -rf`

`-print0`/`-0` will handle spaces in the paths if needed.

The current approach will get slower and slower as the project grows, since it's file by file. The risk of my suggested way is the command length could grow beyond what the shell allows and would get truncated (I've actually had this happen in another context!). But in that case, we'd look at a different approach than either of these anyway.

Reply:
-print0 might not work as intended as it also include the src path and all other leading to the __pycache__ paths, which result in a delete of the entire src/ folder. I'll probably find a fix to it.

Reply:
Weird... I don't remember `-print0` behaving this way on `find`, but I am definitely seeing it. So, yeah, that obviously won't work.

Reply:
Though, we may not need to deal with spaces in file paths for our team. So, could just not worry about it.

Also, the project is small enough, and likely will be for some time, so what you have is okay for now.

Reply:
Ah... got it... order matters for primaries.
`find src -type d -name "__pycache__" -print0` behaves like we want.